### PR TITLE
chore(contract): Update README and dependencies

### PR DESCRIPTION
**Pull request by shuhuiluo**
Reorganize dependencies by moving development-related packages to `devDependencies`. Enhance the README for better readability with structured formatting and shell command snippets, and add a `.prettierignore` file to exclude certain directories from Prettier formatting.

Introduced a new `.solhint.json` configuration file and updated key dependencies in `package.json`. Improved code formatting rules in `.prettierrc.js` and unified style for conditional deployer address assignment in deployment scripts. Fixed a linting command issue in `makefile`.

### CODE REVIEW

**Comment by texuf** [comment]
Nice

File: contracts/.solhint.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "solhint:default",
```
Comment (giuseppecrj) [inline contracts/.solhint.json:2]: There's already a solhint.json at the root of the project. https://github.com/river-build/river/blob/main/.solhint.json

File: contracts/.prettierrc.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
+    plugins: ['prettier-plugin-solidity'],
```
Comment (giuseppecrj) [inline contracts/.prettierrc.js:2]: can you extend from our current prettier-config and add this as an extension

File: contracts/.prettierignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+cache/
```
Comment (giuseppecrj) [inline contracts/.prettierignore:1]: there's already a prettierignore at the root of the project

**Review by giuseppecrj — CHANGES_REQUESTED** [review]

File: contracts/.prettierrc.js (hunk 2 above)
Comment (shuhuiluo) [reply contracts/.prettierrc.js]: Done.

File: contracts/.prettierignore (hunk 3 above)
Comment (shuhuiluo) [reply contracts/.prettierignore]: Gotcha.

File: contracts/.solhint.json (hunk 1 above)
Comment (shuhuiluo) [reply contracts/.solhint.json]: Have to make some changes to pass CI.

**Review by giuseppecrj — APPROVED** [review]

**Pull request merged**